realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -1818,4 +1822,40 @@ public void testDeletingRealmAlsoClearsConfigurationCache() throws IOException {
         testRealm = Realm.getInstance(newConfig);
         assertNotNull(testRealm);
     }
+
+    public void testProcessLocalListenersAfterRefresh() {
+        // Used to validate the result
+        final AtomicBoolean listenerWasCalled = new AtomicBoolean(false);
+
+        // Used by the background thread to wait for the main thread to do the write operation
+        final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
Perhaps call this `bgThreadReady` or something. `latch` is a bit too generic imo.